### PR TITLE
T lo/build dev packages

### PR DESCRIPTION
**Pull request by t-lo**
This change introduces `build_dev_binpkgs`, a script to build binary packages for all dependencies of the devcontainer. This works around an issue with `build_packages`, which doesn't - leading to build issues with the devcontainer later on. This particularly happens for more complex builds with the devcontainer.

Additionally, a call to `build_dev_binpkgs` has been added to the package publishing step in ci-automation before binary packages are published.

Lastly, this change adds a command line option to `build_packages` to stop execution of the script after breaking circular dependencies (i.e. before board packages are built). This option was very helpful while working on `build_dev_binpkgs` and is likely to be similarly useful for other package build related development.

Fixes https://github.com/flatcar/Flatcar/issues/1292.

## How to use

```shell
./run_sdk_container -t
./build_packages
./build_dev_binpkgs
```

## Testing done

Successfully built devcontainer packages for amd64 and arm64.

# Backporting and cherry-picking

Should be backported / cherry-picked to all maintenance branches: flatcar-3033, flatcar-3510, flatcar-3602, flatcar-3760, flatcar-3815.

### CODE REVIEW

**Comment by pothos** [comment]
Started a Jenkins build: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/3111/cldsv/

**Comment by t-lo** [comment]
Restarted the Jenkins CI after fixing an unbound variable issue: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/3120/cldsv/

**Comment by t-lo** [comment]
Libtool is back!
* amd64: https://bincache.flatcar-linux.net/boards/amd64-usr/3817.0.0+devpackages/pkgs/sys-devel/
* https://bincache.flatcar-linux.net/boards/arm64-usr/3817.0.0+devpackages/pkgs/sys-devel/

**Comment by t-lo** [comment]
All tests are green on Jenkins CI, and updated packages now include libtool once again: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/3126/cldsv/

File: build_dev_binpkgs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+if [[ $# -eq 0 ]]; then
+  set -- "${packages_default[@]}"
+fi
+# --
+
+function my_board_emerge() {
+	export PORTAGE_CONFIGROOT="/build/${FLAGS_board}"
+	export SYSROOT="${SYSROOT:-/build/${FLAGS_board}}"
+	export ROOT="/build/${FLAGS_board}"
+	sudo -E emerge --root-deps=rdeps "${@}"
```
Comment (pothos) [inline build_dev_binpkgs:52]: The export is global, how about moving it to the invocation - this has the same effect for the command but does not leak to the function caller
```suggestion
	PORTAGE_CONFIGROOT="/build/${FLAGS_board}" SYSROOT="${SYSROOT:-/build/${FLAGS_board}}" ROOT="/build/${FLAGS_board}" sudo -E emerge --root-deps=rdeps "${@}"
```

Comment (pothos) [reply build_dev_binpkgs:52]: one can also do `sudo VAR=val cmds…` and drop the `-E` if it's only about these vars and not others

**Review by pothos — APPROVED** [review]

**Comment by t-lo** [comment]
Merged and cherry-picked for flatcar-3815.

3033, 3510, 3602 and 3760 required backports so these are dealt with in separate PRs.

**Pull request merged**